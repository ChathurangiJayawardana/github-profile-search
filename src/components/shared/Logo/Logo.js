import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import colors from '../../../setup/colors';

const Logo = props => (
  <Container {...props}>
    <span className="logo-github">Github</span>
    <span className="logo-search">Search</span>
  </Container>
);

const Container = styled.div`
  background: ${colors.white};
  width: 415px;
  height: 81px;
  font-size: 60px;

  ${props => props.headerMode && css`
    width: 280px;
    height: 50px;
    font-size: 40px;
  `}

  .logo-github {
    font-family: 'Monaco', sans-serif;
    margin-right: 5px;
  }

  .logo-search {
    font-family: 'Raleway', sans-serif;
    font-weight: 200;
    font-style: italic;
  }
`;

Logo.propTypes = {
  headerMode: PropTypes.bool,
};

Logo.defaultProps = {
  headerMode: false,
};

export default Logo;

