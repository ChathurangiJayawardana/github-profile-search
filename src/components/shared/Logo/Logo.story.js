import React from 'react';
import { storiesOf } from '@storybook/react';
import Logo from '.';

const stories = storiesOf('shared/Logo', module);

stories
  .add('default', () => <Logo />)
  .add('headerMode', () => <Logo headerMode />);

