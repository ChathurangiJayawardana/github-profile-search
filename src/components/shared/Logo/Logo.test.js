import React from 'react';
import { mount, shallow } from 'enzyme';

import Logo from '.';

describe('Logo Component', () => {
  describe('renders', () => {
    it('render without props ', () => {
      const wrapper = shallow(<Logo />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('content', () => {
    it('should contain 2 span', () => {
      const wrapper = mount(<Logo />);

      expect(wrapper.find('span')).toHaveLength(2);
    });
  });
});

