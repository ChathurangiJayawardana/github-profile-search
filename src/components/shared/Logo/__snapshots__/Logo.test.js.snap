// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component renders render without props  1`] = `ShallowWrapper {}`;
